crates/bevy_rerecast_core/src/generator/mom_can_we_have_weak_handle.rs

```diff
@@ -0,0 +1,48 @@
+use bevy_asset::{StrongHandle, prelude::*};
+use std::hash::{Hash, Hasher};
+use std::sync::{Arc, Weak};
+
+/// We need this lovely construct because
+/// - We want to store `Weak<StrongHandle>` as our index, since we can upgrade it to a strong handle when needed.
+///   - We don't store `Handle` directly because we don't want to keep a strong reference to the asset.
+///   - Keeping the asset alive is the user's choice, not the library's.
+///   - We cannot store an `AssetId` instead because inserting into a dropped asset ID will cause Bevy to hit "unreachable" code lol
+/// - But `Weak<StrongHandle>` is not `Eq`!!!
+///   - We could use a `Vec<(A, B)>` instead of a `HashMap`, but the code in [`NavmeshGenerator::regenerate`] also wants an eq check.
+///
+/// So, we build this bad boi ourselves :bavy:
+#[derive(Debug, Clone)]
+pub(crate) struct WeHaveWeakHandleAtHome<T: Asset> {
```

Comment:
Maybe we should call this `UpgradableAssetId` or something?

Reply:
Good idea! https://github.com/janhohenheim/rerecast/pull/19/commits/c70daf3a4b441fe10d4b179e409edc0cba9859a9